### PR TITLE
TT-1601 Integration tests ensure that the MSA can generate test tool …

### PR DESCRIPTION
**Pull request by RachaelBooth**
…json

Should ensure that the json interface the test tool uses stays in sync
with the matching service adapter.
Tests that the json used in each of the default tests in the resources
folder of the test tool can be produced by the matching service adapter.
N.B. Does not ensure that the schema which validates user supplied test
cases remains up to date.

Authors: @rachaelbooth

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -60,10 +60,6 @@ configurations {
 }
 
 configurations.all {
-    resolutionStrategy {
-        // dropwizard-testing uses 2.2.0 but our libraries currently only work with 1.6.0.
```
Comment (RachaelBooth) [inline build.gradle]: N.B. This comment lies. Pre-commit passes.

**Comment by RachaelBooth** [comment]
I've tried to change the test cases in the test tool as little as possible - just the pids and the ordering of the addresses (the MSA always puts the current address first, and now has a test for that...).

File: src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+                ), false, REQUEST_ID)), REQUEST_ID, HUB_ENTITY_ID, PID))
+            .build();
+
+        Path path = Paths.get("verify-matching-service-test-tool/src/main/resources/LoA2-extensive-case.json");
+
+        assertThatRequestThatWillBeSentIsEquivalentToFile(attributeQuery, path);
+    }
+
+    @Test
+    public void shouldProducesUserAccountCreationJson() throws Exception {
```
Comment (andy-paine) [inline src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java]: shouldProduce_s_? ⌨️

File: src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+    private static final String REQUEST_ID = "default-match-id";
+    private static final String PID = "default-pid";
+    private static final ObjectMapper objectMapper = Jackson.newObjectMapper().setDateFormat(ISO8601DateFormat.getDateInstance());
+    private static final Integer replaceYesterday = 1;
+    private static final Integer replace405to100 = 100;
+    private static final Integer replace405to101 = 150;
+    private static final Integer replace405to200 = 400;
+    private static final Integer replace180to100 = 100;
+    private static final Integer replace180to101 = 140;
+    private static final Integer replace180to150 = 160;
```
Comment (andy-paine) [inline src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java]: Replace is a slightly odd name as it looks like we'll be replacing 180 with 150 but the value is 160. Maybe something like `inRange180to150`?

File: src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+
+    @ClassRule
+    public static WireMockClassRule wireMockRule = new WireMockClassRule(wireMockConfig().port(1234));
```
Comment (andy-paine) [inline src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java]: I think you can do `wireMockConfig().dynamicPort()` here instead and then use `wireMockRule.getPort()` or something similar in the ConfigOverride URLs

Comment (RachaelBooth) [reply src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java]: I get a class initialisation error from the app rule when I do that, sadly. Done a bit more googling, but still not found a way round it, so I think we do have to do the fixed port thing.

**Pull request merged**